app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,100 @@
+class InvitationsController < ApplicationController
+  before_action :restrict_access_on_create, only: [ :new, :create ]
+  before_action :restrict_access_on_update, only: [:edit, :update]
+
+  # GET /invitations/new
+  def new
+    @user = User.new
+  end
+
+  # POST /invitations
+  def create
+    @user = User.new(invitation_params)
+
+    # Fictive params to save new user
+    @user.update(name: "name", surname: "surname", gender: :male, 
+                 password: "password", password_confirmation: "password")
+
+    if (@user.save)
+      invitation_token = new_token
+      @user.invitation_digest = digest invitation_token
+      @user.activated = false;
+      @user.save
+
+      # Mail goes here
+      @token = invitation_token
+      puts edit_invitation_url(invitation_token, email: @user.email)
+      render file: "user_mailer/invitation.text.erb"
+
+      # Real mail
+      # UserMailer.invitation(invitation_token, @user).deliver
+    else
+      render "new"
+    end
+  end
+
+  # GET /invitations/token/edit
+  def edit
+    @user = User.find_by(email: params[:email])
+
+    unless (@user && !@user.activated? && 
+            BCrypt::Password.new(@user.invitation_digest).is_password?(params[:id]))
+      flash[:alert] = "Invalid activation link!"
+      redirect_to root_path
+    end
+
+    @user.name = ""
+    @user.surname = ""
+  end
+
+  # PUT /invitations/token
+  def update
+    @user = User.find_by(email: params[:user][:email])
+
+    if @user.update(activation_params)
+      @user.update(activated: true)
+      sign_in(:user, @user)
+      redirect_to root_path
+    else
+      render "edit"
+    end
+  end
+
+private
+  # Only administrators have permition to invite new users 
+  def restrict_access_on_create
+    unless (user_signed_in? && current_user.admin?)
+      flash[:alert] = "You should be logged in as administrator."
+      redirect_to new_user_session_path
+    end
+  end
+
+  # Authorized users shouldn't activate new account's
+  def restrict_access_on_update
+    if (user_signed_in?)
+      flash[:alert] = "You are signed in."
```

Comment:
Why? What is point of this alert?
Imagine a user who has just **signed up** and you notify him that he is **signed in**. I mean, what is really purpose of telling him? He was signing up (not signing in) while you tell about the latter.
And still, even if you change it to `signed up`, what's point?